mobile/views/DirectMessageComposer/Composer.js

```diff
@@ -0,0 +1,214 @@
+      const { wipSearchString } = this.state;
+      if (wipSearchString && wipSearchString.length < 5) {
+        this.searchThrottled(wipSearchString);
+      } else {
+        this.searchDebounced(wipSearchString);
+      }
+    });
+  };
+
+  onFinishTyping = (e: { nativeEvent: { text: string } }) => {
+    this.search(e.nativeEvent.text);
+  };
+
+  searchDebounced = (searchString: ?string) => {
+    this.setState({
+      searchString,
+    });
+  };
+
+  searchThrottled = (searchString: ?string) => {
+    this.setState({
+      searchString,
+    });
+  };
+
+  search = (searchString: ?string) => {
+    this.setState({
+      searchString,
+    });
+  };
+
+  selectUser = (userId: string) => {
+    this.setState(prev => ({
+      selectedUsers: prev.selectedUsers.concat([userId]),
+      wipSearchString: '',
+      searchString: '',
+    }));
+    this.searchInput && this.searchInput.focus();
+  };
+
+  removeSelectedUser = (userId: string) => () => {
+    this.setState(prev => ({
+      selectedUsers: prev.selectedUsers.filter(
+        selectedId => selectedId !== userId
+      ),
+    }));
+  };
+
+  onSubmit = (text: string) => {
+    if (this.state.selectedUsers.length === 0) return;
+    this.props
+      .createDirectMessageThread({
+        participants: this.state.selectedUsers,
+        message: {
+          messageType: 'text',
+          threadType: 'directMessageThread',
+          content: {
+            body: text,
+          },
+        },
+      })
+      .then((result: CreateDirectMessageThreadType) => {
+        const { navigation } = this.props;
+
+        return navigation.navigate('DirectMessageThread', {
+          id: result.data.createDirectMessageThread.id,
+        });
+      })
+      .catch(err => {
+        console.error(err);
+      });
+  };
+
+  filterResults = (results: Array<Object>) => {
+    const { currentUser } = this.props;
+    const { selectedUsers } = this.state;
+
+    return results
+      .filter(row => row.id !== currentUser.id)
+      .filter(row => selectedUsers.indexOf(row.id) < 0);
+  };
+
+  render() {
+    return (
+      <ComposerWrapper>
+        <SearchInputArea>
+          <SelectedUsers
+            horizontal
+            empty={this.state.selectedUsers.length === 0}
+          >
+            {this.state.selectedUsers.map(userId => (
+              <SelectedUser
+                key={userId}
+                id={userId}
+                keyboardShouldPersistTaps={'always'}
+                onPressHandler={this.removeSelectedUser(userId)}
+              />
+            ))}
+          </SelectedUsers>
+          <UserSearchInput
+            onChangeText={this.onChangeText}
+            onEndEditing={this.onFinishTyping}
+            onSubmitEditing={this.onFinishTyping}
+            value={this.state.wipSearchString}
+            returnKeyType="search"
+            autoFocus
+            innerRef={elem => (this.searchInput = elem)}
+            placeholder={'Search for people...'}
```

Comment:
nit: no autocorrect and autocapitalize

Reply:
Do we want those on user search? Autocorrect may be more annoying than not when searching for names

Reply:
That's why I said _no_ autocorrect and autocapitalize—right now it autocorrects for me.